src/uuid_utils.ts

```diff
@@ -8,26 +8,26 @@ const VALIDATION_REGEX =
 export const uuidValidateString = (str: string): boolean =>
   typeof str === 'string' && VALIDATION_REGEX.test(str);
 
-export const uuidHexStringToBuffer = (hexString: string): Buffer => {
+export const uuidHexStringToBuffer = (hexString: string): Uint8Array => {
   if (!uuidValidateString(hexString)) {
     throw new BSONTypeError(
       'UUID string representations must be a 32 or 36 character hex string (dashes excluded/included). Format: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" or "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx".'
     );
   }
 
   const sanitizedHexString = hexString.replace(/-/g, '');
-  return Buffer.from(sanitizedHexString, 'hex');
+  return ByteUtils.fromHex(sanitizedHexString);
 };
 
-export const bufferToUuidHexString = (buffer: Buffer, includeDashes = true): string =>
-  includeDashes
-    ? buffer.toString('hex', 0, 4) +
-      '-' +
-      buffer.toString('hex', 4, 6) +
-      '-' +
-      buffer.toString('hex', 6, 8) +
-      '-' +
-      buffer.toString('hex', 8, 10) +
-      '-' +
-      buffer.toString('hex', 10, 16)
-    : buffer.toString('hex');
+export function bufferToUuidHexString(buffer: Uint8Array, includeDashes = true): string {
+  if (includeDashes) {
+    return [
```

Comment:
I'm happy to revert this change, I made it when there was a bunch of type errors in the initial refactoring of all the byteUtils, but it's kinda clean this way 🤷🙂